### ISSUE TITLE
Stream upload videos 

### ISSUE THREAD

**Issue by nikotz** [Nekmo/telegram-upload#59]
* telegram-upload version: 0.32
* Python version: 3.7
* Operating System: macos/win

### Description

When i upload a video whit a telegram-bot large or small the option of stream it doesnt seem to be activate, and the users have to download the video(mp4 whit encode h264)
![Captura de pantalla 2020-09-18 a la(s) 3 36 05 p  m](https://user-images.githubusercontent.com/6304322/93649683-29ebf780-f9d2-11ea-9050-5b6af30ec6ce.png)
 
searching in telethon seems to be a option for stream ho is for default off maybe i have to turn on or add a flag or make a is a problem whit the app 
![image](https://user-images.githubusercontent.com/6304322/93650099-3a50a200-f9d3-11ea-92f8-dad5461ae3ed.png)

thank you very much for your incredible app

**Comment by Nekmo** [comment]
Hello, please upload the video to the issue for testing. Thanks.

**Comment by nikotz** [comment]
[#conradosev.mp4.zip](https://github.com/Nekmo/telegram-upload/files/5287534/conradosev.mp4.zip)

**Comment by Nekmo** [comment]
I have uploaded this video using `telegram-upload` and it plays in streaming.

```
(telegram-upload) [nekmo@homura ~/Workspace/telegram-upload] (master)$ python telegram_upload/management.py upload /tmp/conradosev.mp4/#conradosev.mp4 
Uploading "#conradosev.mp4"  [####################################]  100% 
```

![Screenshot_20200927_021535](https://user-images.githubusercontent.com/740106/94352622-9ff1fd80-0067-11eb-90d5-258e5c746939.png)

![image](https://user-images.githubusercontent.com/740106/94352633-d0399c00-0067-11eb-8a45-f0ec371cdb85.png)

**Comment by nikotz** [comment]
meaby is the version of my telegram-upload? let me update and try

**Comment by nikotz** [comment]
I upgrade the telegram-upload an upload videos for 200 mb 50 mb and 10 mb and nothing stream
Telegram-upload --to channel  "source.mp4"

**Comment by nikotz** [comment]
This is my channel if you want to send a file, maybe https://t.me/instatolive is open

**Comment by YulinChan** [comment]
The same. The video can not streaming

**Comment by Nekmo** [comment]
Is ffmpeg available in your OS @nikotz ?

**Comment by nikotz** [comment]
yes, in the mac and the windows

**Comment by Nekmo** [comment]
@nikotz Check that hachoir can get video metadata. Open a terminal, run `python` y paste:

```
from telegram_upload.video import video_metadata
print(video_metadata('<video path>'))
```

For example:
```
>>> from telegram_upload.video import video_metadata
>>> print(video_metadata('/home/nekmo/t_video5794143716776085444.mp4'))
Metadata:
- Duration: 3 sec 154 ms
- Image width: 1080 pixels
- Image height: 1920 pixels
- Creation date: 2020-04-08 01:12:55
- Last modification: 2020-04-08 01:12:55
- Comment: Play speed: 100.0%
- Comment: User volume: 100.0%
- MIME type: video/mp4
- Endianness: Big endian
```

**Comment by nikotz** [comment]
Metadata:
- Duration: 37 min 50 sec 751 ms
- Image width: 720 pixels
- Image height: 1396 pixels
- Creation date: 1904-01-01 00:00:00
- Last modification: 1904-01-01 00:00:00
- Comment: Play speed: 100.0%
- Comment: User volume: 100.0%
- MIME type: video/mp4
- Endianness: Big endian
[Finished in 3.7s]

Metadata:
- Duration: 3 hours 20 min 18 sec
- Image width: 721 pixels
- Image height: 406 pixels
- Creation date: 1904-01-01 00:00:00
- Last modification: 1904-01-01 00:00:00
- Comment: Play speed: 100.0%
- Comment: User volume: 100.0%
- MIME type: video/mp4
- Endianness: Big endian
[Finished in 12.0s]

**Comment by Nekmo** [comment]
Ok, metadatas works ... it might be thumbnail issue in OS X/Windows. Try to get a thumbnail using Telegram-upload:

```
from telegram_upload.video import get_video_thumb
get_video_thumb('<video path>')
```

For example:
```
from telegram_upload.video import get_video_thumb
get_video_thumb('/home/nekmo/t_video5794143716776085444.mp4')
'/tmp/tmpz8x2csg3.jpg'
```

Thanks.

**Comment by nikotz** [comment]
says finish but dot know where put the file...

**Comment by nikotz** [comment]
whit this code putting the output, make it but if the output is no defined not generate the file

from telegram_upload.video import get_video_thumb
get_video_thumb('/Users/nicolastenorio/Downloads/cam4/2020.09.04_14.35.57_abie_owen.mkv.mp4.mp4', '5.jpg')

![5](https://user-images.githubusercontent.com/6304322/94749714-16ce1580-034a-11eb-9566-2e7c45e9663d.jpg)

**Comment by YulinChan** [comment]
telethon can upload streamable video
```
from telethon import TelegramClient, sync

client = TelegramClient("send_video", 123456, '345d2da90bd9b1124e359')
client.start()
client.send_file('@lhzaobao', 'test.mp4', caption='test', supports_streaming=True)
```
<img width="396" alt="Snipaste_2020-10-03_13-39-48" src="https://user-images.githubusercontent.com/48314862/94984456-abb95480-057e-11eb-8508-eabc4eb138bf.png">

**Comment by Nekmo** [comment]
@nikotz @YulinChan I have been able to confirm the bug in the latest releases of Telethon. In my version `Telethon==1.8.0` the streaming does work.

**Comment by nikotz** [comment]
> telethon can upload streamable video
> 
> ```
> from telethon import TelegramClient, sync
> 
> client = TelegramClient("send_video", 123456, '345d2da90bd9b1124e359')
> client.start()
> client.send_file('@lhzaobao', 'test.mp4', caption='title', supports_streaming=True)
> ```
> 
> <img alt="Snipaste_2020-10-03_13-39-48" width="396" src="https://user-images.githubusercontent.com/48314862/94984456-abb95480-057e-11eb-8508-eabc4eb138bf.png">

this work for me, @Nekmo how can upgrade telethon

**Comment by Nekmo** [comment]
@nikotz try the latest development release from Telegram-upload:

```
pip install https://github.com/Nekmo/telegram-upload/archive/master.zip
```

Thanks

**Comment by nikotz** [comment]
i cant install the upgrade 

```
Requirement already satisfied (use --upgrade to upgrade): telegram-upload==0.3.3 from https://github.com/Nekmo/telegram-upload/archive/master.zip in /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages
Requirement already satisfied: telethon in /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages (from telegram-upload==0.3.3) (1.16.4)
Requirement already satisfied: click>=6.0 in /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages (from telegram-upload==0.3.3) (7.1.2)
Requirement already satisfied: cryptg in /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages (from telegram-upload==0.3.3) (0.2.post1)
Requirement already satisfied: hachoir in /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages (from telegram-upload==0.3.3) (3.1.1)
Requirement already satisfied: rsa in /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages (from telethon->telegram-upload==0.3.3) (4.6)
Requirement already satisfied: pyaes in /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages (from telethon->telegram-upload==0.3.3) (1.6.1)
Requirement already satisfied: pycparser in /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages (from cryptg->telegram-upload==0.3.3) (2.20)
Requirement already satisfied: cffi>=1.0.0 in /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages (from cryptg->telegram-upload==0.3.3) (1.14.2)
Requirement already satisfied: pyasn1>=0.1.3 in /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages (from rsa->telethon->telegram-upload==0.3.3) (0.4.8)
```

**Comment by nikotz** [comment]
HI i downgrade the telethon to version 1.8 and i think it worked

**Comment by Nekmo** [comment]
@nikotz in the latest release of Telegram-upload there is support for the latest version of Telethon, you can update both:

```
pip install https://github.com/Nekmo/telegram-upload/archive/master.zip
pip install -U Telethon
```

**Comment by nikotz** [comment]
@Nekmo , i have the last relasse and no work, but i think the problem is an update of telethon i try all the version since 1.8 and the last version worked i 1.11.3 since 1.12 stop working

**Comment by Nekmo** [comment]
@nikotz can you check the packages you have installed on your system?

```
(telegram-upload) [nekmo@homura ~/Workspace/telegram-upload] (master)$ pip freeze
acronym==1.8.2
alabaster==0.7.12
async-generator==1.10
asyncqt==0.8.0
Babel==2.8.0
bleach==3.2.1
bump2version==1.0.0
bumpversion==0.6.0
certifi==2020.6.20
cffi==1.14.3
chardet==3.0.4
click==7.1.2
colorama==0.4.3
cryptg==0.2.post1
cryptography==3.1.1
docutils==0.16
hachoir==3.1.1
idna==2.10
imagesize==1.2.0
importlib-metadata==2.0.0
jeepney==0.4.3
Jinja2==2.11.2
joblib==0.17.0
keyring==21.4.0
MarkupSafe==1.1.1
nltk==3.5
numpy==1.19.2
packaging==20.4
pbr==5.5.0
Pillow==7.2.0
pkginfo==1.5.0.1
pyaes==1.6.1
pyasn1==0.4.8
pycparser==2.20
Pygments==2.7.1
pyparsing==2.4.7
PySide2==5.15.1
pytz==2020.1
readme-renderer==26.0
regex==2020.9.27
requests==2.24.0
requests-toolbelt==0.9.1
rfc3986==1.4.0
rsa==4.6
SecretStorage==3.1.2
shiboken2==5.15.1
six==1.15.0
snowballstemmer==2.0.0
Sphinx==3.2.1
sphinx-click==2.5.0
sphinxcontrib-applehelp==1.0.2
sphinxcontrib-devhelp==1.0.2
sphinxcontrib-htmlhelp==1.0.3
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.3
sphinxcontrib-serializinghtml==1.1.4
telegram-upload==0.3.3
Telethon==1.16.4
tqdm==4.50.0
twine==3.2.0
urllib3==1.25.10
webencodings==0.5.1
zipp==3.3.0
You are using pip version 9.0.1, however version 20.2.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
(telegram-upload) [nekmo@homura ~/Workspace/telegram-upload] (master)$ 
```

**Comment by nikotz** [comment]
altgraph==0.10.2
appdirs==1.4.3
backports-abc==0.5
backports.functools-lru-cache==1.6.1
bdist-mpkg==0.5.0
beautifulsoup4==4.9.0
bonjour-py==0.3
certifi==2019.11.28
cffi==1.14.0
chardet==3.0.4
click==7.1.1
colorama==0.4.3
cryptography==2.9.1
decorator==4.4.2
dirhunt==0.7.0
enum34==1.1.10
future==0.18.2
futures==3.3.0
geoip2==3.0.0
gevent==1.4.0
google==3.0.0
greenlet==0.4.15
humanize==1.0.0
idna==2.9
ipaddress==1.0.23
livestreamer==1.12.2
macholib==1.5.1
matplotlib==1.3.1
maxminddb==1.5.2
modulegraph==0.10.4
numpy==1.8.0rc1
proxy-db==0.2.5
py2-ipaddress==3.4.2
py2app==0.7.3
pyaes==1.6.1
pyasn1==0.4.8
pycparser==2.20
pyobjc-core==2.5.1
pyobjc-framework-Accounts==2.5.1
pyobjc-framework-AddressBook==2.5.1
pyobjc-framework-AppleScriptKit==2.5.1
pyobjc-framework-AppleScriptObjC==2.5.1
pyobjc-framework-Automator==2.5.1
pyobjc-framework-CFNetwork==2.5.1
pyobjc-framework-Cocoa==2.5.1
pyobjc-framework-Collaboration==2.5.1
pyobjc-framework-CoreData==2.5.1
pyobjc-framework-CoreLocation==2.5.1
pyobjc-framework-CoreText==2.5.1
pyobjc-framework-DictionaryServices==2.5.1
pyobjc-framework-EventKit==2.5.1
pyobjc-framework-ExceptionHandling==2.5.1
pyobjc-framework-FSEvents==2.5.1
pyobjc-framework-InputMethodKit==2.5.1
pyobjc-framework-InstallerPlugins==2.5.1
pyobjc-framework-InstantMessage==2.5.1
pyobjc-framework-LatentSemanticMapping==2.5.1
pyobjc-framework-LaunchServices==2.5.1
pyobjc-framework-Message==2.5.1
pyobjc-framework-OpenDirectory==2.5.1
pyobjc-framework-PreferencePanes==2.5.1
pyobjc-framework-PubSub==2.5.1
pyobjc-framework-QTKit==2.5.1
pyobjc-framework-Quartz==2.5.1
pyobjc-framework-ScreenSaver==2.5.1
pyobjc-framework-ScriptingBridge==2.5.1
pyobjc-framework-SearchKit==2.5.1
pyobjc-framework-ServiceManagement==2.5.1
pyobjc-framework-Social==2.5.1
pyobjc-framework-SyncServices==2.5.1
pyobjc-framework-SystemConfiguration==2.5.1
pyobjc-framework-WebKit==2.5.1
pyOpenSSL==0.13.1
pyparsing==2.0.1
python-dateutil==1.5
python-telegram-bot==12.6.1
pytz==2013.7
requests==2.23.0
requests-mock==1.8.0
rsa==4.5
scipy==0.13.0b1
singledispatch==3.4.0.3
six==1.4.1
soupsieve==1.9.5
SQLAlchemy==1.3.16
telegram-send==0.20
Telethon==0.18.2.3
tornado==5.1.1
typing==3.7.4.3
urllib3==1.25.8
websocket==0.2.1
websocket-client==0.57.0
xattr==0.6.4
youtube-dl==2020.9.6
zope.interface==4.1.1

**Comment by Nekmo** [comment]
@nikotz  maybe you need to uninstall telegram-upload and reinstall it.

```
pip uninstall telegram-upload
pip install https://github.com/Nekmo/telegram-upload/archive/master.zip
```

I have installed your dependencies and reinstalling it works. Remember to install from Github: this version is not yet available on Pypi.

Regards

**Comment by nikotz** [comment]
@Nekmo Thank you very much for your patience i uninstall telegram-upload and telethon and installl the last update and is working

**Comment by Nekmo** [comment]
> @Nekmo Thank you very much for your patience i uninstall telegram-upload and telethon and installl the last update and is working

You are welcome, thank you for your time and for testing this new release!

It will soon be available on Pypi.

**Comment by Nekmo** [comment]
Released version 0.3.4 Update it using:

```
pip install -U telegram-upload
```
